aggregator_core/src/datastore.rs

```diff
@@ -4630,6 +4630,38 @@ ON CONFLICT(task_id, batch_identifier, aggregation_param) DO UPDATE
         )
     }
 
+    /// Delete an `aggregate_share_job` row from the datastore.
+    #[tracing::instrument(skip(self), err(level = Level::DEBUG))]
+    pub async fn delete_aggregate_share_job(
```

Comment:
What's the story for deleting things like batch aggregations or report aggregations? Do we expect that garbage collection will eventually clean them up?

Reply:
`delete_expired_collection_artifacts` will definitely still get them, as it's keyed off the TaskID instead.

This is where I have a weak understanding still of the data model. The `batch_aggregations` table has `batch_identifier` onto which we can cascade a delete from the aggregate share jobs.

The `report_aggregations` table connects to the `aggregation_jobs` table which has `batch_id`. Would you expect the delete to cascade through all three tables, @tgeoghegan ?
- batch_aggregations, via batch_aggregations.batch_identifier
- aggregation_jobs, via aggregation_jobs.batch_id
- report_aggregations, via aggregation_jobs.aggregation_job_id